make/Makefile.iso

```diff
@@ -70,33 +70,27 @@ $(DESTINATION)/tree.tar.zst:
 	make create-repo
 
 .PHONY: local-iso
-local-iso: $(LUET) $(MAKEISO) $(DESTINATION) $(MKSQUASHFS) $(DESTINATION)/tree.tar.zst
+local-iso: $(ELEMENTAL) $(DESTINATION) $(MKSQUASHFS) $(DESTINATION)/tree.tar.zst
 ifneq ($(shell id -u), 0)
 	@echo "*** Please run 'make $@' as root"
 	@exit 1
 endif
 ifneq ("$(ISO)","")
 	@echo "'$(ISO) exists, run 'make clean_iso' folled by 'make $@' to recreate"
 else
-	$(LUET) makeiso -- $(MAKEISO_ARGS) $(MANIFEST) --local $(DESTINATION)
```

Comment:
I think logs are misleading here, I am not sure from where this log is coming from but this is likely to be the config loaded from the system. Then before installing any packages custom repositories are used if provided. Provided repositories are not appended to any existing repositories list, the provided repositories are the only ones being used. There is no mechanism to append repositories to the system wide ones, you either use whatever is in the system or the runtime provided list.

Reply:
`--repo` flag (which can be repeated, the result is a string slice) will initiate a slice of `v1.Repository` which then in will be passed to [pkg/luet/luet.go](https://github.com/rancher-sandbox/elemental/blob/ed32b2e85140abdb2217e46740804bc237aca4be/pkg/luet/luet.go#L189). The command `build-iso` will NEVER use the repositories defined in the system such as in `/etc/luet/repos.d`, if repos are not provided within the manifest (where you can define priority and other repo paramters) or within the command line using `--repo` flag, build-iso command will only use the default repository.

IMHO from an image builder perspective makes no sense reading the repositories form the system as this couples the build to the running host, which should never happen, the builds should be consistent regardless the host they are built on. So that repositories are explicitly provided (by the user or by elemental defaults).

Reply:
> then I'm not sure if this might work. Before we were appending the local repository to the remote one, increasing the priority of the local. This is in order to let luet resolve packages from remote but take the newest from the local repo with the changes at hand

There could be a case in which the local build and the remote build are having the same version then the criteria about which to pick cannot be considered deterministic.

@mudler What about this behavior:

* default repo has a non `0` priority, something that is not the maximum, `10`, `90`... whatever.
* `--repo` flag instead of replacing default repository it appends on top of default repository or whatever is defined within the `manifest.yaml`. 

This way we can easily add a high priority repository over a defined list (by default or by manifest) and we still can completely use different repositories (no defaults) by setting a repository list in manifest.

Gonna make a PR for that.